lib/portlayer/exec/commit.go

```diff
@@ -246,6 +216,91 @@ func Commit(op trace.Operation, sess *session.Session, h *Handle, waitTime *int3
 	return nil
 }
 
+// batchBlockOnFunc is a batching routine that batch processes incoming requests.
+// Incoming request signals the batching routine by throwing into a batch channel. When this routine is performing operation on
+// the previous requests, new requests will be batched in the channel, waiting for next iteration to process.
+func batchBlockOnFunc(ctx context.Context, batch chan chan error, operation func(operation trace.Operation) error) {
+	op := trace.FromContext(ctx, "Add container VM to VMGroup dispatch routine")
+
+	for {
+		members := make([]chan error, 0, 5) // batching queue
+		var req chan error
+		var ok bool
+
+		// block and wait for first request
+		select {
+		case req, ok = <-batch:
+			if !ok {
+				return
+			}
+			if req == nil {
+				continue
+			}
+			members = append(members, req)
+		case <-op.Done(): // when parent context is cancelled, quit
+			return
+		}
+
+		// fetch batched requests
+		for {
+			if len(batch) == 0 { // nothing batched for now, quit looping
```

Comment:
Is there any reason not to just put this condition in the `for`? (i.e., `for len(batch) > 0 {`)

Reply:
it would infinite loop on `len(batch) == 0`. It enters the loop no matter what, if length of batch doesn't meet the condition, since it's in the loop, it's gonna loop anyway. Also we want to keep the condition check within the for loop, since the batch size is changing

Reply:
> It enters the loop no matter what

I don't think this is how Golang handles `for` loops.

Consider this [example](https://play.golang.org/p/zF1sbOAT-m_F):
```
func main() {
    i := 0

    for i > 0 {
        fmt.Println("BAD!")
        break
    }
}
```

It will exit without printing "BAD!".

Reply:
ohhh I see what you mean! thanks. fixing it now